### ISSUE TITLE
Don't treat header tags as landmarks unless they're children of the body tag

### ISSUE THREAD

**Issue by sergiofenoll** [ember-template-lint/ember-template-lint#2259]
Hi!

I ran up against an issue today where the [no-duplicate-landmark-elements](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-duplicate-landmark-elements.md) rule treats every single `<header>` tag as a landmark (i.e., having the banner role).

According to what I can find on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header#accessibility) and in the [HTML spec](https://w3c.github.io/html-aria/#el-header), `<header>` tags get `role=banner` only [when they're the scoped to the `<body>` tag](https://w3c.github.io/html-aam/#el-header-ancestorbody), in [other cases they shouldn't be](https://w3c.github.io/html-aam/#el-header).

In short I believe that the following code should get a warning from the no-duplicate-landmark-elements rule

```
<body>
  <header></header>
  <header></header>
</body>
```

but this code shouldn't, but it currently does
```
<section>
  <header></header>
  <header></header>
</section>
```
